### PR TITLE
Add boot-loader for apps

### PR DESCRIPTION
**Pull request by felixheidecke**
Apps ship with `boot.js` to register themselves and navigation elements.

### CODE REVIEW

**Comment by felixheidecke** [comment]
This is the interesting part:

https://github.com/owncloud/phoenix/blob/app-loader/apps/market/js/boot.js

File: config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (DeepDiver1975) [inline config.json]: we didn't want to check this one in. please adopt the sample config file

File: index.html [hunk 2]
```diff
@@ -107,7 +107,11 @@
 		</div>
 	</main>
 
-	<script src="core/uikit/dist/js/uikit.min.js"></script>
-	<script src="core/js/core.bundle.js" defer></script>
+    <script src="core/uikit/dist/js/uikit.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/vue/2.5.16/vue.js"></script>
```
Comment (DeepDiver1975) [inline index.html]: cdn? why? nogo. think of offline instances and about privacy

File: config.json.sample [hunk 3]
```diff
@@ -1,18 +1,6 @@
 {
 	"backendUrl": "http://localhost/owncloud",
-	"apps": {
```
Comment (DeepDiver1975) [inline config.json.sample]: why drop apps? why own apps.json? One config to rule them all - right?

You wanted to reduce requests ;-)

File: src/default.js [hunk 4]
```diff
@@ -34,14 +39,16 @@ OC = new Vue({
...
-				this.apps = apps;
-				deferred.resolve();
-			});
-
-			return deferred;
+			$.when(
+				$.getJSON('/config.json', (config) => {
+					this.config = config;
+				}),
+				$.getJSON('/apps.json', (apps) => {
```
Comment (DeepDiver1975) [inline src/default.js]: why two requests?

File: src/default.js [hunk 5]
```diff
@@ -52,26 +59,31 @@ OC = new Vue({
 		 */
 
 		_setupApps () {
-			var deferred = $.Deferred();
 
-			_.map(this.apps, (app, name) => {
+			_.forEach(this.apps, (app, i) => {
 
-				requirejs([`${this.appPath}/${name}/js/boot.js`], ( boot ) => {
+				// TODO: Find better var name for 'foo'
```
Comment (DeepDiver1975) [inline src/default.js]: bar?

**Pull request merged**